viper/parser/stmt.py

```diff
@@ -275,5 +276,51 @@ def parse_return(self):
                 setter = make_setter(new_sub, sub, 'memory')
                 return LLLnode.from_list(['seq', setter, ['return', new_sub, get_size_of_type(self.context.return_type) * 32]],
                                             typ=None, pos=getpos(self.stmt))
+
+        # Returning a tuple.
+        elif isinstance(sub.typ, TupleType):
+
+            subs = []
+            dynamic_offset_counter = LLLnode(self.context.get_next_mem(), typ=None, annotation="dynamic_offset_counter")  # dynamic offset position counter.
+            new_sub = LLLnode.from_list(self.context.get_next_mem() + 32, typ=self.context.return_type, location='memory', annotation='new_sub')
+            keyz = list(range(len(sub.typ.members)))
+            dynamic_offset_start = 32 * len(sub.args)  # The static list of args end.
+            left_token = LLLnode.from_list('_loc', typ=new_sub.typ, location="memory")
+
+            def get_dynamic_offset_value():
+                # Get value of dynamic offset counter.
+                return ['mload', dynamic_offset_counter]
+
+            def increment_dynamic_offset(dynamic_spot):
+                # Increment dyanmic offset counter in memory.
+                return ['mstore', dynamic_offset_counter,
+                                 ['add',
+                                        ['add', ['ceil32', ['mload', dynamic_spot]], 32],
+                                        ['mload', dynamic_offset_counter]]]
+
```

Comment:
Do you think splitting the `LLL` into functions makes the code easier to read?  I personally like it together in chunks, but may be biased because that's what I'm used to doing.

Reply:
@DavidKnott in this specific case splitting out which each one does and giving it a name helped me a lot. Usually I use `annotation=` but couldn't really here because we are building the with statement. Simple cases using a bunch of LLL definitely work for me, but I like describing complex cases like the dynamic output here as much as possible (mostly for my feature self).

Reply:
Thanks for sharing your thoughts, I'll play around more with how I write my `LLL`.